### PR TITLE
boolean operators

### PR DESCRIPTION
**Pull request by hammad13060**
proposal for ```logical and```, ```logical or``` and ```logical not```.

### CODE REVIEW

File: test/expression-language.yml [hunk 1]
```diff
@@ -28,6 +28,66 @@ context: {a: 3.0, b: 2.0}
 source: 'b / a'
 result: 0.6666666666666666
 ---
+title: 'logical not (1)'
+context: {key1: true, key2: false}
```
Comment (jonasfj) [inline test/expression-language.yml]: I would rename the context `{tt: true, ff: false}`
That way the truth table becomes a lot easier to read...

File: src/interpreter.js [hunk 2]
```diff
@@ -131,6 +131,12 @@ let testMathOperand = (operator, operand) => {
   return;
 };
 
+let testLogicalOperand = (operator, operand) => {
+  if (!isBool(operand)) {
+    throw expectationError(`infix: ${operator}`, `boolean ${operator} boolean`);
```
Comment (jonasfj) [inline src/interpreter.js]: We probably should pass in `token` instead of `operator`...
Notably we want the position information from the `token` if we're going to throw an error... we probably should try to include position information in `InterpreterError`. But we can also do that in a separate PR.

Just a note.

**Comment by jonasfj** [comment]
Feel free to hit merge.

File: src/interpreter.js (hunk 2 above)
Comment (hammad13060) [inline src/interpreter.js]: We should do this in a separate PR because we can use ```syntaxerror.js``` as a common error class. what do you say @jonasfj ?
